src/msg/producer/writer/shard_writer.go

```diff
@@ -160,6 +160,11 @@ func newReplicatedShardWriter(
 
 func (w *replicatedShardWriter) Write(rm *producer.RefCountedMessage) {
 	w.RLock()
+	if len(w.messageWriters) == 0 {
+		w.m.noWritersError.Inc(1)
+		w.RUnlock()
+		return
+	}
```

Comment:
The `noWritersError` actually belongs to `messageWriter`:

https://github.com/m3db/m3/blob/9daa1e499757ab8468b345632bc1e8e861f1ab06/src/msg/producer/writer/message_writer.go#L459-L463

It gets emitted when using `sharedShardWriter` when calling `messageWriter::Write()` and the `messageWriter` is empty. 

https://github.com/m3db/m3/blob/9daa1e499757ab8468b345632bc1e8e861f1ab06/src/msg/producer/writer/shard_writer.go#L77-L79

This doesn't work `replicatedShardWriter` since `w.messageWriters` array can be empty and then no `messageWriter::Write()` is called.

https://github.com/m3db/m3/blob/9daa1e499757ab8468b345632bc1e8e861f1ab06/src/msg/producer/writer/shard_writer.go#L161-L167